src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -324,19 +324,27 @@ def add_pipeline_api_route(app: FastAPI, pipeline_name: str, pipeline_wrapper: B
 
 
 def deploy_pipeline_files(
-    app: FastAPI, pipeline_name: str, files: dict[str, str], save_files: bool = True, overwrite: bool = False
+    pipeline_name: str,
+    files: dict[str, str],
+    app: Optional[FastAPI] = None,
```

Comment:
why `app` has become optional?

Reply:
Basically, `app` is needed if we want to add dynamic pipeline-specific API routes to the FastAPI `app`. This is true when we deploy a pipeline from CLI (or using `/deploy-files` endpoint), but not if we deploy a pipeline using the deploy MCP Tool (in that case you will not have a FastAPI `app` instance).

Said that, I was thinking about having a single server with both MCP and REST API endpoints (especially now that we have streamable http as a transport). In that case, a deployed pipeline will add to the server both a new API route and a new MCP Tool.

Reply:
Thanks for the explanation!